ui/app/models/keymgmt/provider.js

```diff
@@ -0,0 +1,119 @@
+import Model, { attr } from '@ember-data/model';
+import { tracked } from '@glimmer/tracking';
+import { expandAttributeMeta } from 'vault/utils/field-to-attrs';
+import { validator, buildValidations } from 'ember-cp-validations';
+
+const CRED_PROPS = {
+  azurekeyvault: ['client_id', 'client_secret', 'tenant_id'],
+  awskms: ['access_key', 'secret_key', 'session_token', 'endpoint'],
+  gcpckms: ['service_account_file'],
+};
+const OPTIONAL_CRED_PROPS = ['session_token', 'endpoint'];
+// since we have dynamic credential attributes based on provider we need a dynamic presence validator
+// add validators for all cred props and return true for value if not associated with selected provider
+const credValidators = Object.keys(CRED_PROPS).reduce((obj, providerKey) => {
+  CRED_PROPS[providerKey].forEach((prop) => {
+    if (!OPTIONAL_CRED_PROPS.includes(prop)) {
+      obj[`credentials.${prop}`] = validator('presence', {
+        presence: true,
+        value(model) {
+          return model.credentialProps.includes(prop) ? model.credentials[prop] : true;
+        },
+      });
+    }
+  });
+  return obj;
+}, {});
+const Validations = buildValidations({
+  name: validator('presence', true),
+  keyCollection: validator('presence', true),
+  ...credValidators,
+});
+const ValidationsModel = Model.extend(Validations);
+
+export default class KeymgmtProviderModel extends ValidationsModel {
+  @attr('string', {
+    label: 'Provider name',
+    subText: 'This is the name of the provider that will be displayed in Vault. This cannot be edited later.',
+  })
+  name;
+
+  @attr('string', {
+    label: 'Type',
+    subText: 'Choose the provider type.',
+    possibleValues: ['azurekeyvault', 'awskms', 'gcpckms'],
+    defaultValue: 'azurekeyvault',
+  })
+  provider;
+
+  @attr('string', {
+    label: 'Key Vault instance name',
+    subText: 'The name of a Key Vault instance must be supplied. This cannot be edited later.',
+  })
+  keyCollection;
+
+  @attr('date') created;
+
+  idPrefix = 'provider/';
+  type = 'provider';
+
+  @tracked keys = [];
+  @tracked credentials = null; // never returned from API -- set only during create/edit
+
+  get icon() {
+    return {
+      azurekeyvault: 'azure-color',
+      awskms: 'aws-color',
+      gcpckms: 'gcp-color',
+    }[this.provider];
+  }
+  get typeName() {
+    return {
+      azurekeyvault: 'Azure Key Vault',
+      awskms: 'AWS Key Management Service',
+      gcpckms: 'Google Cloud Key Management Service',
+    }[this.provider];
+  }
+  get showFields() {
+    const attrs = expandAttributeMeta(this, ['name', 'created', 'keyCollection']);
+    attrs.splice(1, 0, { hasBlock: true, label: 'Type', value: this.typeName, icon: this.icon });
+    const l = this.keys.length;
+    const value = l ? `${l} ${l > 1 ? 'keys' : 'key'}` : 'None';
```

Comment:
We actually have a `type=array` option on InfoTableRow which may allow you to not to this fussing for the keys, at least. See `transform-role-edit.hbs` for one example of how that works.

Reply:
Cool that's good to know! The `InfoTableItemArray` component won't work in this case since a number based on the array length needs to be displayed such as _6 keys_ rather than the actual array elements.